design/src/rfcs/rfc0027_new_home_for_cred_cache_and_builder_api_consistency.md

```diff
@@ -0,0 +1,182 @@
+- `ConfigLoader` has `credentials_provider` and `credentials_cache`, but that pair of methods does not seem to behave consistently in `aws_types::sdk_config::Builder` and every service client's `config::Builder`.
+- A credentials cache can only be created within the `aws-config` crate.
+
+Proposed Solutions
+------------------
+In this section we will talk about our proposed solution separately for each of the bullet points from the previous section, one for API consistency and one for creating credentials caches.
+
+### API consistency for `credentials_provider` and `credentials_cache` in other builders
+We propose to have `sdk_config::Builder` and service clients' `config::Builder`s match `ConfigLoader` and take the `credentials_cache` method to make it consistent all the way through.
+
+- :+1: The user can expect a pair of `credentials_provider` and `credentials_cache` methods behaves consistently across `ConfigLoader`,  `sdk_config::Builder` and service clients' `config::Builder`s.
+- :-1: It will require some duplicate code across those places, although it's rather an insignificant amount.
+
+### Creating credentials caches outside the `aws-config` crate
+We propose to move types and traits for credentials caching to a separate crate (tentatively called `aws-XXX`). Specifically, the following items will be moved to `aws-XXX`:
+- All items in `aws_types::credentials` and their dependencies
+- All items in `aws_config::meta::credentials` and their dependencies
+
+The new crate will be the home for `ProvideCachedCredentials` discussed above as well as `CredentialsCache` with its `Lazy` variant and builder. Crucially, `CredentialsCache::create_cahe` will be marked as `pub`. The `aws-XXX` crate itself depends on `aws_smithy_types` and `aws_smithy_async`, among others, for the items in it to be able to compile. There is one wrinkle, however, that is `LazyCredentialsCache` depends on `aws_types::os_shim_internal::TimeSource`; it does not feel right to move `TimeSource` to `aws-XXX` just because `LazyCredentialsCache` requires it. There are two ways to work around it.
+
+#### Option A: Moving `aws_types::sdk_config` to `aws-XXX`
+
+A result of this arrangement will give us the following module interdependencies:
+<p align="center">
+  <img width="800" alt="Option A" src="https://user-images.githubusercontent.com/15333866/206587781-6eca3662-5096-408d-a435-d4023929e727.png">
+</p>
+
+- :+1: `aws-XXX::sdk_config::Builder` and a service client `config::Builder` can create a `SharedCredentialsCache` with a concrete type of credentials cache.
+- :+1: It avoids cyclic crate dependencies.
+- :-1: There is one more crate to maintain and version.
+- :-1: `aws-XXX::sdk_config` sitting together with the types and traits for credentials provider & caching may not give us a coherent mental model for the `aws-XXX` crate, making it difficult to choose the right name for `XXX`.
+
+#### Option B: Moving `aws_types::sdk_config` to yet another new crate
```

Comment:
I'm not sure I fully understand option B. Does the second crate only have `SdkConfig` in it or something?

Reply:
8c5ce64 removed option B as it was marginally different from option A, which is now the proposed solution.

Reply:
Updated the selected design as per our discussion and what was previously option A is now considered a dismissed design in fb28d8c.